src/ril_binder_radio.c

```diff
@@ -4249,18 +4751,39 @@ ril_binder_radio_arg_name(
         RIL_BINDER_DEFAULT_NAME);
 }
 
+static
+enum radio_interface
+ril_binder_radio_arg_interface(
+    GHashTable* args)
+{
+    const char *name = ril_binder_radio_arg_value(args,
+        RIL_BINDER_KEY_INTERFACE, RIL_BINDER_DEFAULT_INTERFACE);
```

Comment:
There's no need to pass a non-NULL string as a default only to find it on the list and return a enum value (BTW why `enum radio_interface` and not `RADIO_INTERFACE`?). Get rid of RIL_BINDER_DEFAULT_INTERFACE, pass NULL, and return the default enum value if no interface is specified:
```c
static
RADIO_INTERFACE
ril_binder_radio_arg_interface(
    GHashTable* args)
{
    const char *name = ril_binder_radio_arg_value(args,
        RIL_BINDER_KEY_INTERFACE, NULL);

    if (name) {
        RADIO_INTERFACE i;

        for (i = RADIO_INTERFACE_1_0; i < RADIO_INTERFACE_COUNT; i++) {
            if (!g_strcmp0(name, ril_binder_radio_interface_name(i))) {
                return i;
            }
        }
    }
    return DEFAULT_INTERFACE;
}
```
There's no need to print anything here, especially as a WARNING, the selected value will by printed by `ril_binder_radio_init_base()`. And there's no need for `ril_binder_radio_interface_supported()` check, the logic in `ril_binder_radio_init_base()` should handle any value, supported or not.